daemon/guestos_mgr.c

```diff
@@ -371,30 +397,47 @@ push_config_verify_buf_cb(smartcard_crypto_verify_result_t verify_result, unsign
 	//			     GUESTOS_MGR_UPDATE_DOWNLOAD_NO_WIFI);
 
 	// 4. trigger image download if os is used by a container or a fresh install
-	if (guestos_mgr_is_guestos_used_by_containers(os_name) || !old_os) {
+	if (guestos_mgr_is_guestos_used_by_containers(os_name) || !old_os || cml_update) {
 		INFO("%s: %s", GUESTOS_MGR_UPDATE_TITLE, GUESTOS_MGR_UPDATE_DOWNLOAD);
-		guestos_mgr_download_latest(os_name);
+		if (control_send_message(CONTROL_RESPONSE_GUESTOS_MGR_INSTALL_STARTED, *resp_fd) < 0)
+			WARN("Could not send response to fd=%d", *resp_fd);
+		guestos_mgr_download_latest(os_name, *resp_fd);
 	}
+
+	mem_free(resp_fd);
 	return;
 
 cleanup_purge:
 	guestos_purge(os);
 cleanup_os:
 	guestos_free(os);
+err:
+	if (control_send_message(CONTROL_RESPONSE_GUESTOS_MGR_INSTALL_FAILED, *resp_fd) < 0)
+		WARN("Could not send response to fd=%d", *resp_fd);
+	mem_free(resp_fd);
```

Comment:
freeing data pointer always desired?

Reply:
when cb is done it has to free its associated data yes.